### PR TITLE
Update ipk to v0.5.1

### PR DESCRIPTION
**Pull request by blinard-BIOINFO**
Updates IPK to v0.5.1, which introduces several bugfixes.

### CODE REVIEW

**Comment by blinard-BIOINFO** [comment]
@BiocondaBot please add ipk

**Comment by blinard-BIOINFO** [comment]
@Juke34 sorry for the long delay to correct this, the rule is now passing tests for version update

**Comment by Juke34** [comment]
@BiocondaBot please fetch artifacts

**Review by Juke34 — CHANGES_REQUESTED** [review]
It would have been nice that the main executable was called using `ipk` instead of `ipk.py`.
But if you do not care about that detail I think we are good to go.
FYI the package is working for OSX intel but not arm.

File: recipes/ipk/build.sh [hunk 1]
```diff
@@ -13,16 +13,24 @@ mkdir -p $PREFIX/bin
 mkdir -p $PREFIX/lib
 
 cmake -B build -DHASH_MAP=USE_TSL_ROBIN_MAP -DCMAKE_CXX_FLAGS="-O3" -DBUILD_SHARED_LIBS=ON --install-prefix=$PREFIX
-cmake --build build --target all
+cmake --build build --target all --target diff-dna --target diff-aa --target dump-dna --target dump-aa
```
Comment (martin-g) [inline recipes/ipk/build.sh:16]:
```suggestion
cmake --build build --target all --target diff-dna --target diff-aa --target dump-dna --target dump-aa -j ${CPU_COUNT}
```

File: recipes/ipk/build.sh [hunk 2]
```diff
@@ -13,16 +13,24 @@ mkdir -p $PREFIX/bin
...
 cmake --install build
 
-ls build/ipk
+ls $PREFIX/*
```
Comment (martin-g) [inline recipes/ipk/build.sh:19]: Debug statement that could be removed

File: recipes/ipk/build.sh [hunk 3]
```diff
@@ -13,16 +13,24 @@ mkdir -p $PREFIX/bin
...
-chmod +x $PREFIX/bin/ipk-aa-pos
-chmod +x $PREFIX/bin/ipk.py
+cp build/i2l/libi2l_aa.a $PREFIX/lib
+cp build/i2l/libi2l_aa_pos.a $PREFIX/lib
+cp build/i2l/libi2l_dna.a $PREFIX/lib
+
+#chmod +x $PREFIX/bin/ipk-aa
+#chmod +x $PREFIX/bin/ipk-dna
+#chmod +x $PREFIX/bin/ipk-aa-pos
+#chmod +x $PREFIX/bin/ipk.py
```
Comment (martin-g) [inline recipes/ipk/build.sh:36]: The commented out lines could be removed

Comment (blinard-BIOINFO) [reply recipes/ipk/build.sh]: done

File: recipes/ipk/build.sh (hunk 2 above)
Comment (blinard-BIOINFO) [reply recipes/ipk/build.sh]: done

**Comment by blinard-BIOINFO** [comment]
I did you requested changes. For technical reason, right now we will stick to ipk.py, making it evolve to a single binary is in our todo list for future releases.

File: recipes/ipk/build.sh (hunk 3 above)
Comment (blinard-BIOINFO) [reply recipes/ipk/build.sh]: done

**Review by martin-g — APPROVED** [review]

**Review by Juke34 — APPROVED** [review]

**Comment by blinard-BIOINFO** [comment]
@martin-g @Juke34 
I'm getting a weird behavior related to boost following the package development.

In the conda-build CI, package `boost-cpp=1.85` got selected for the conda env  (the rule requires `boost-cpp>=1.67`).
As one can see here:

```
# see pipeline https://github.com/bioconda/bioconda-recipes/runs/27086383834

ONDA INFO (OUT) The following NEW packages will be INSTALLED:
15:55:23 BIOCONDA INFO (OUT) 
15:55:23 BIOCONDA INFO (OUT)     _libgcc_mutex:    0.1-conda_forge   conda-forge
15:55:23 BIOCONDA INFO (OUT)     _openmp_mutex:    4.5-2_gnu         conda-forge
15:55:23 BIOCONDA INFO (OUT)     boost-cpp:        1.85.0-h44aadfe_2 conda-forge        <<===
15:55:23 BIOCONDA INFO (OUT)     bzip2:            1.0.8-hd590300_5  conda-forge
15:55:23 BIOCONDA INFO (OUT)     icu:              73.2-h59595ed_0   conda-forge
15:55:23 BIOCONDA INFO (OUT)     libboost:         1.85.0-hba137d9_2 conda-forge
15:55:23 BIOCONDA INFO (OUT)     libboost-devel:   1.85.0-h00ab1b0_2 conda-forge
15:55:23 BIOCONDA INFO (OUT)     libboost-headers: 1.85.0-ha770c72_2 conda-forge

[...]

CONDA INFO (OUT) -- Found Boost: $PREFIX/lib/cmake/Boost-1.85.0/BoostConfig.cmake (found version "1.85.0") found components: serialization iostreams system filesystem

```

So the binary is built with boost v1.85 on bioconda side.
However, when installing on a fresh env in my local computer, conda or mambe refuses to use `boost-cpp` 1.85 and stick to boost-cpp 1.84. This results to a conflit of shared boost librairies and the program fails to run.

```
# running the command 

ipk.py build --refalign reference.fasta --reftree tree.rooted.newick --states nucl --workdir . --model GTR

/home/belinard/SOFTWARE_LOCAL/miniconda3/envs/epik/bin/ipk-dna: error while loading shared libraries: libboost_program_options.so.1.85.0: cannot open shared object file: No such file or directory

# the binary from the package look for v1.85 like in conda-build

ldd miniconda3/envs/epik/bin/ipk-dna
	linux-vdso.so.1 (0x00007ffeebbd7000)
	libboost_program_options.so.1.85.0 => not found
	libboost_filesystem.so.1.85.0 => not found
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x0000788eb5097000)
	libboost_serialization.so.1.85.0 => not found
	libboost_iostreams.so.1.85.0 => not found
	libstdc++.so.6 => /home/belinard/SOFTWARE_LOCAL/miniconda3/envs/epik/bin/../lib/libstdc++.so.6 (0x0000788eb4eb2000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x0000788eb4dcb000)
	libgcc_s.so.1 => /home/belinard/SOFTWARE_LOCAL/miniconda3/envs/epik/bin/../lib/libgcc_s.so.1 (0x0000788eb4dac000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000788eb4a00000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x0000788eb4da5000)
	/lib64/ld-linux-x86-64.so.2 (0x0000788eb515b000)

# but the command `conda install ipk epik` installs v1.84

# conda list gives:

boost-cpp                 1.84.0               h44aadfe_3    conda-forge
libboost                  1.84.0               hba137d9_3    conda-forge
libboost-devel            1.84.0               h00ab1b0_3    conda-forge
libboost-headers          1.84.0               ha770c72_3    conda-forge

# i can confirm by looking at the files
miniconda3/envs/epik/lib/libboost_serialization.so

```

I looked at the repo libboost-feedstock and it seems this can happen if channel_priority is not set to strict for conda-forge.
I check the documentation, but so far did not found a way to force "strict" priority for a package. Is this even possible ?

I will set the booost-cpp version to a strict version in the recipe. 
But I wanted to let you know that this weird behaviour happens (and may break potential other libboost-based packages, which may be super common for C++ packages).

**Pull request merged**